### PR TITLE
fix: add CollectionNormalizerTest && ItemNormalizerTest && ApiDocumentationBuilderTest

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: tests/Hydra/Serializer/ItemNormalizerTest.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+use Symfony\Component\Serializer\Normalizer\AbstractNormalizer;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ItemNormalizerTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @expectedException \ReflectionException
```
Comment (dunglas) [inline tests/Hydra/Serializer/ItemNormalizerTest.php]: It's weird isn't it?

File: tests/Hydra/Serializer/ItemNormalizerTest.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        $propertyAccess = $this->prophesize(PropertyAccessorInterface::class);
+        $nameConverter = $this->prophesize(NameConverterInterface::class);
+
+        $normalizer = new ItemNormalizer($resourceMetadataFactoryProphecy->reveal(), $propertyNameCollectionFactoryProphecy->reveal(), $propertyMetadataFactoryProphecy->reveal(), $iriConverterProphecy->reveal(), $resourceClassResolverProphecy->reveal(), $contextBuilderProphecy->reveal(), $propertyAccess->reveal(), $nameConverter->reveal());
+        $normalizer->setSerializer($serializer->reveal());
+
+        $expected = ['@context' => '/contexts/Dummy',
+                     '@id' => '/dummies/1988',
+                     '@type' => 'http://schema.org/Thing',
+                     '' => null,
```
Comment (dunglas) [inline tests/Hydra/Serializer/ItemNormalizerTest.php]: Should be removed IMO.

File: tests/Hydra/Serializer/ItemNormalizerTest.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+
```
Comment (dunglas) [inline tests/Hydra/Serializer/ItemNormalizerTest.php]: Extra blank lines to remove.

**Comment by dunglas** [comment]
I left some minor comments! Thank you very much for improving our test coverage.

**Comment by Simperfit** [comment]
@dunglas comments addressed

**Comment by Simperfit** [comment]
@dunglas ApiDocumentationBuilder Test added.

**Comment by Simperfit** [comment]
@dunglas ApiDocumentationBuilder Test added.

**Comment by dunglas** [comment]
Thank you @Simperfit

**Pull request merged**